VAMobile/src/components/FormWrapper/FormWrapper.test.tsx

```diff
@@ -63,112 +59,85 @@ context('FormWrapper', () => {
       onSaveClicked = value
     })
 
-    component = render(
+    render(
       <FormWrapper
         fieldsList={fieldsList}
         onSave={onSaveSpy}
-        setFormContainsError={() => {}}
+        setFormContainsError={() => { }}
         resetErrors={resetErrors}
         setOnSaveClicked={setOnSaveClicked}
         onSaveClicked={onSaveClicked}
       />,
     )
-
-    testInstance = component.UNSAFE_root
   }
 
   beforeEach(() => {
     initializeTestInstance()
   })
 
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
-  })
-
   describe('when the picker calls setError with an empty string', () => {
-    it('should set the error to empty string', async () => {
-      await waitFor(() => {
-        let shortenedFieldsList = formFieldsList[1]
-        initializeTestInstance([shortenedFieldsList])
-        testInstance.findByType(VAModalPicker).props.setError('')
-        expect(screen.queryByText('second error message')).toBeFalsy()
-      })
+    it('should set the error to empty string', () => {
+      initializeTestInstance(pickerFieldList)
+      screen.UNSAFE_getByType(VAModalPicker).props.setError('')
```

Comment:
there should be another way to get the VAModalPicker instead of this unsafe way that we are trying to get rid of

Reply:
Reading through the docs the reason those methods are called `UNSAFE_` is only in reference to how you have to have implicit knowledge of the props and types that are being passed, which we do here, that is the point, we've got to access that prop to test the error handling, so I'm not convinced that in this instance we should try to get rid of them...

Reply:
Looking at React Testing Library's [Guiding Principles](https://testing-library.com/docs/guiding-principles/):

> If it relates to rendering components, then it should deal with DOM nodes rather than component instances, and it should not encourage dealing with component instances.

RTL's general philosophy is to test components as a user would see/interact with them, and to avoid testing implementation details whenever possible. The more implementation details you interact with, the more fragile the tests become because refactoring can break the tests even if the functionality is still complete and correct from the user's perspective.

I feel like these types of tests, "when the picker calls setError with an empty string" it "should set the error to empty string" are somewhat in the weeds of implementation details. Is there a way the user would interact with this component that would test these types of scenarios in a more complete way, without digging into the implementation details? If so that would be more in line with the RTL approach.

Reply:
I get that, and I agree, but from what I understand (granted, I might be wrong) the Wrapper basically just wraps the form fields passed into it and does validation and error handling for the form.... I'm happy to remove all the "sets the correct errors" tests, but that doesn't leave much _to_ test. 

As for how the user interacts with it, the user doesn't necessarily interact directly with this component wrapper, only the child form components (except maybe the test for the "reset errors")... @Sparowhawk and I spoke about it and agreed that while these are `UNSAFE_` methods, and yes, we're getting a little too into the weeds with testing, if we want test coverage on this component, we need to test the functionality, and this was the only way I could see to do that. :( 

I'm certainly open to suggestions if you want to poke at it and come up with a better way to test it though!!

Reply:
I looked at this some more, and I think the best approach would be to set the child form components to have invalid or empty values, and simulate the saved state for the form. That should cause FormWrapper to show the error messages. The final test in this file, just before the commented-out stuff, has some of the right approach. It's not exactly right since it is setting props directly, but it does simulate the state after onSave is called